tests/myapp/tests.py

```diff
@@ -1806,3 +1806,51 @@ def test_insert_ordered_BFS_backwards_nonroot_nodes(self):
             2 1 1 1 4 7
             4 2 1 2 5 6
         """)
+
+
+class CacheChildrenTestCase(TreeTestCase):
+    """
+    Tests that the queryset function `get_cached_trees` results in a minimum
+    number of database queries.
+    """
+    fixtures = ['categories.json', 'genres.json']
+
+    def test_category_iter(self):
+        """
+        Test a query with one root node.
+        """
+        with self.assertNumQueries(1):
+            root_nodes = Category.objects.all().get_cached_trees()
+
+        # `get_cached_trees` should only return the root nodes
+        self.assertEqual(len(root_nodes), 1)
+
+        # Getting the ancestors of each node should not result in db hits.
+        with self.assertNumQueries(0):
+            self.assertEqual(len(root_nodes), 1)
+            category = root_nodes[0]
+            self.assertIsInstance(category, Category)
+            self.assertEqual(len(category.get_children()), 3)
+            for child in category.get_children():
+                self.assertIsInstance(child, Category)
+                for child in category.get_children():
+                    self.assertIsInstance(child, Category)
+
+    def test_genre_iter(self):
```

Comment:
Does this test a different edge case than the other test?

If not, maybe just have one of them :)